src/umpire/resource/NoOpMemoryResource.cpp

```diff
@@ -0,0 +1,67 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/resource/NoOpMemoryResource.hpp"
+#include <cstdint>
+#include <stdlib.h>
+#include <string.h>
+#include "umpire/ResourceManager.hpp"
+#include "umpire/util/Macros.hpp"
+
+namespace umpire {
+namespace resource {
+
+
+NoOpMemoryResource::NoOpMemoryResource(Platform platform,
+                                       const std::string& name, int id,
+                                       MemoryResourceTraits traits)
+    : MemoryResource{name, id, traits}, m_platform{platform}
+{
+  m_count = (UINT64_C(1)<<48);
```

Comment:
This may be done C++ 11 style and just moved to the place it was declared in the header file (see below).

```suggestion
```